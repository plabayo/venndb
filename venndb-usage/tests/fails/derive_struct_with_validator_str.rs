use venndb::VennDB;

#[derive(Debug, VennDB)]
#[venndb(validator = "employee_validator")]
struct Employee {
    pub id: u32,
    pub name: String,
    pub is_manager: bool,
    pub is_admin: bool,
    pub is_active: bool,
    pub department: Department,
}

#[derive(Debug)]
pub enum Department {
    Engineering,
    Sales,
    Marketing,
    HR,
}

fn employee_validator(employee: &Employee) -> bool {
    employee.id > 0 && !employee.name.is_empty()
}

fn main() {
    let _ = EmployeeDB::new();
}
