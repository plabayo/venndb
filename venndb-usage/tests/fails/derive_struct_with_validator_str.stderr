error: Expected path attribute, found "employee_validator" attribute (literal)
 --> tests/fails/derive_struct_with_validator_str.rs:4:22
  |
4 | #[venndb(validator = "employee_validator")]
  |                      ^^^^^^^^^^^^^^^^^^^^
